.github/workflows/go.yml

```diff
@@ -46,3 +46,15 @@ jobs:
 
       - name: git diff
         run: git diff --exit-code
+
+  go_tidy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: "go.mod"
+      - run: |
+          go mod tidy
+          git diff --exit-code
```

Comment:
(nit) Please split these into separate steps so we have a visual confirmation of `git diff` actually running.

Reply:
I don't think that's really necessary, but, why not, done 😉